tests/net_dynamic_hb.rs

```diff
@@ -108,11 +111,7 @@ fn do_drop_and_re_add(cfg: TestConfig) {
                 id
             );
             let dhb = DynamicHoneyBadger::builder().build(node.netinfo.clone());
-            SenderQueue::builder(
-                dhb,
-                node.netinfo.all_ids().filter(|&&them| them != id).cloned(),
-            )
-            .build(node.id)
+            SenderQueue::builder(dhb, node.netinfo.all_ids().cloned()).build(node.id)
```

Comment:
This might work because `SenderQueue` filters out its own ID internally. But conceptually the IDs here are all IDs other than the ID of the node for which the `SenderQueue` is being built.

Reply:
I'm not sure about that: Isn't it usually convenient to be able to just pass all the network's IDs to the constructor/builder?
If we decide against that, should we add a check instead, and make the builder fail if it contains our own ID?

(I also found some branches in the sender queue code that looked like we were expected to actually have an entry for our own ID…)

Reply:
I'm not sure about convenience. Our epoch is known from `algo.epoch()`. You would duplicate it by including the pair of our ID and our epoch into `peer_epochs`.

Reply:
That's true. But to my mind, that means we should either filter our own ID, or fail if we find it. And the sender queue shouldn't contain code that checks for our own ID in the map.

Reply:
`peer_epochs` is updated on a message by the `SenderQueue`. On the contrary, `self.algo.epoch` is updated internally by the algorithm. That's why I think these two things should not overlap and we should not duplicate own epoch by including it in `peer_epochs`.

Reply:
Yes, I completely agree. The question is just how we prevent it:
* Should the constructor/builder panic if it sees our own ID among the peer IDs?
* Or should it just filter our won ID from them?

Reply:
An assertion in the constructor/builder would be great. After that, if own ID appears in `peer_epochs` that must be due to user code sending a message with the sender being us. That's arguably a bug in the user code. Could be a panic too, what do you think?

Reply:
I still feel it would be a more convenient API to allow the user to just pass in all node IDs (and automatically filter out the own ID). E.g. imagine you're implementing a blockchain client and there's a chain spec configuring the initial set of validators. It would be convenient to be able to just pass that list into the `SenderQueue` constructor. I'd actually guess that in the majority of use-cases, the user would have to do this manual filtering if we add that check.

Anyway, it's definitely unrelated to the actual test fix, so I reverted my changes for now. We can modify the `SenderQueueBuilder` in a separate PR.